api/types/authentication.go

```diff
@@ -290,6 +281,9 @@ func (c *AuthPreferenceV2) CheckAndSetDefaults() error {
 	if c.Spec.DisconnectExpiredCert == nil {
 		c.Spec.DisconnectExpiredCert = NewBoolOption(false)
 	}
+	if c.Origin() == "" {
+		c.SetOrigin(OriginDynamic)
+	}
```

Comment:
This origin value check might more properly belong to `setStaticFields` as well...? If so, then the code should also be updated for `SessionRecordingConfig` and `ClusterNetworkingConfig`.

Reply:
I'm going to keep it limited to fully static fields (all resources of that type have the same value). `setStaticFields` will only have logic like `if` statements during version migrations and refactors. Otherwise the distinction between `setStaticFields` and `CheckAndSetDefaults` is unclear.